src/ParametricOptInterface.jl

```diff
@@ -157,11 +167,20 @@ function MOI.supports_constraint(
 ) where {T}
     return MOI.supports_constraint(
         model.optimizer,
+        # JD: This looks odd. It is asking for ScalarQuadraticFunction and returning Affine
```

Comment:
@odow and @joaquimg this implementation is causing a problem because GLPK isn't compatible with `MathOptInterface.ScalarQuadraticFunction` but the `ParametricOptimizer` is compatible with `MathOptInterface.ScalarQuadraticFunction` so when the tests try to add a `MathOptInterface.ScalarQuadraticFunction` for the `ListOfConstraintIndices` test it fails. 

What is the correct way to proceed here? should support return true or false?

Reply:
To add more detail: 

This function: 

```julia
function MOI.add_constraint(
    model::ParametricOptimizer,
    f::MOI.ScalarQuadraticFunction{T},
    set::MOI.AbstractScalarSet,
) where {T}
    if !function_has_parameters(model, f)
        return MOI.add_constraint(model.optimizer, f, set)
    else
        return add_constraint_with_parameters_on_function(model, f, set)
    end
end
```

gets called with a QuadraticTerm that doesn't have parameters in the tests because the current implementation.

Reply:
I don't think we have a good way of distinguishing between constraint types that are always supported and constraints that are sometimes supported depending on the data (convex vs non-convex quadratics are another example). 

We have to say "true" here if we want `parameter * variable` to work. 

Not sure what a good solution is.

Reply:
> Perhaps we should take a step back (wider than this PR, cc @joaquimg):
> 
> How often do we want `parameter * variable`? It's much slower to implement/modify, some solvers won't support, and it complicates all of POI (which would be trivial if it only had to deal with affine functions).

I agree the bulk of the challenges relate to the handling of the quadratic terms. Generally is strange to handle a Quadratic that in fact is an Affine at the solver call.

Reply:
Yeah I think we had a plan that it would work. But not sure how viable it is in practice

Reply:
It's something that could always be added back later. But it seems to be complicating a lot of the initial MOI interface. It might be better to remove for now, get out a rock-solid linear interface. Test and release that. Then consider quadratic. 

Limiting to affine is also not blocking the conic work etc.

Reply:
What's the plan then? Strip all the code for `p*v` and implement RHS properly?

Reply:
@joaquimg needs to chime in first. This is really his project/needed by PSR.

Reply:
I would like to keep parameters * variables.
And we throw an error in case of variable * variable product not supported by inner solver.
Benoit's analogy is pretty good. Xpress will accept quadratic but will break if non-convex.
If we really want to be strict, I propose a parameter for POI for enabling and disabling quadratic.
I would be fine with a `unsafe_support_quadratic_constraints` keyword.